plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/PointColorGradient.tsx

```diff
@@ -1,13 +1,163 @@
 
+const generateConditions = (
+  field?: string,
+  min?: number,
+  max?: number,
+  step?: number,
+  startColor?: string,
+  endColor?: string,
+) => {
+  if (!field || !min || !max || !step || !startColor || !endColor || min >= max || step >= max) {
+    return [];
+  }
+
+  const values = [];
+  for (let i = min; i <= max; i += step) values.push(i);
+  const colors = generateColorGradient(startColor, endColor, values.length);
+  const fieldName = "${" + String(field) + "}";
+  return values
+    .map((value, index) => [`${fieldName} >= ${String(value)}`, `'${String(colors[index])}'`])
+    .reverse();
+};
+
 const PointColorGradient: React.FC<BaseFieldProps<"pointColorGradient">> = ({
   value,
   editMode,
+  isActive,
   onUpdate,
 }) => {
-  // remember to update the BaseFieldProps type!
-  console.log(value, editMode, onUpdate);
-  return <div>PointColor</div>;
+  const [colorGradient, setColorGradient] = useState(value);
+
+  const handleFieldChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setColorGradient({ ...colorGradient, field: e.target.value });
+    },
+    [colorGradient],
+  );
+
+  const handleMinValueUpdate = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const min = !isNaN(parseFloat(e.currentTarget.value)) ? parseFloat(e.currentTarget.value) : 1;
+      setColorGradient({ ...colorGradient, min });
+    },
+    [colorGradient],
+  );
+
+  const handleMaxValueUpdate = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const max = !isNaN(parseFloat(e.currentTarget.value)) ? parseFloat(e.currentTarget.value) : 1;
+      setColorGradient({ ...colorGradient, max });
+    },
+    [colorGradient],
+  );
+
+  const handleStartColorUpdate = useCallback(
+    (color: string) => {
+      // colors should be hex for now
+      if (color && /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(color)) {
+        setColorGradient({ ...colorGradient, startColor: color });
+      }
+    },
+    [colorGradient],
+  );
+
+  const handleEndColorUpdate = useCallback(
+    (color: string) => {
+      // colors should be hex for now
+      if (color && /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(color)) {
+        setColorGradient({ ...colorGradient, endColor: color });
+      }
+    },
+    [colorGradient],
+  );
+
+  const handleStepUpdate = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const step = !isNaN(parseFloat(e.currentTarget.value))
+        ? parseFloat(e.currentTarget.value)
+        : 1;
+      setColorGradient({ ...colorGradient, step });
+    },
+    [colorGradient],
+  );
+
+  const conditions = useMemo(() => {
+    const { field, min, max, step, startColor, endColor } = colorGradient;
+    return generateConditions(field, min, max, step, startColor, endColor);
+  }, [colorGradient]);
+
+  useEffect(() => {
+    if (!isActive || isEqual(omit(colorGradient, "override"), omit(value, "override"))) return;
```

Comment:
Can you update the colorGradient useState to be just the color, not value. Value includes override and type and id, etc.
If you do that, you can more simply and deliberately check if the two are equal